### PR TITLE
feat: (Platform) table phase 2

### PR DESCRIPTION
**Pull request by olshansky**
#### Please check whether the PR fulfills the following requirements

- [ ] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [ ] tests for the changes that have been done
- [ ] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [ ] update `README.md`
- [ ] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [ ] Documentation Examples
- [ ] Stackblitz works for all examples

### CODE REVIEW

File: libs/platform/src/lib/components/table/table.component.ts [hunk 1]
```diff
@@ -80,21 +104,56 @@ interface SelectableRow {
...
         }
     }
-    get dataSource(): FdpTableDataSource<any> {
+    get dataSource(): FdpTableDataSource<T> {
         return this._dataSource;
     }
 
+    /** Initial state of table. */
+    @Input()
+    set state(value: TableState) {
```
Comment (fkolar) [inline libs/platform/src/lib/components/table/table.component.ts:134]: State is one thing to have, but usually its more used to persists user changes but we need to also have a way to pre-initialize DT, when application developer first time add it to the page. 


I have talked about but I dont see it added to the specs and this should be there. 

There needs tobe a bindings 
orderBy , sortBy some similar syntax where we can add colums directly to the table singature

<fdp-table 
            orderBy=[{status, ASC}, {date, DESC}]
     


If there is already a way without instantiating state and adding additional properties, please let me let me know.

File: libs/platform/src/lib/components/table/pipes/value-by-path.pipe.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+import { getNestedValue } from '../../../utils/object';
+
+@Pipe({ name: 'valueByPath', pure: true })
```
Comment (fkolar) [inline libs/platform/src/lib/components/table/pipes/value-by-path.pipe.ts:5]: Pure is by default no? do we need pure:true ?

**Review by fkolar — CHANGES_REQUESTED** [review]

File: libs/platform/src/lib/components/table/pipes/value-by-path.pipe.ts (hunk 2 above)
Comment (dimamarksman) [reply libs/platform/src/lib/components/table/pipes/value-by-path.pipe.ts]: Yeah, you are right, that's truw by default. 
Removed it.

File: libs/platform/src/lib/components/table/table.component.ts (hunk 1 above)
Comment (dimamarksman) [reply libs/platform/src/lib/components/table/table.component.ts:134]: That is a feature of the phase 3.

image

Comment (fkolar) [reply libs/platform/src/lib/components/table/table.component.ts:134]: understood

**Pull request closed without merging**